metrics/internal/collectors/ocs-client.go

```diff
@@ -114,3 +137,22 @@ func (s *storageAutoScalerLister) List(selector labels.Selector) (storageAutoSca
 func NewStorageAutoScalerLister(indexer cache.Indexer) Lister[ocsv1.StorageAutoScaler] {
 	return &storageAutoScalerLister{indexer: indexer}
 }
+
+type OperatorConditionLister interface {
+	List(selector labels.Selector) (ret []*operatorv2.OperatorCondition, err error)
+}
+
+type operatorConditionLister struct {
+	indexer cache.Indexer
+}
+
+func NewOperatorConditionLister(indexer cache.Indexer) OperatorConditionLister {
```

Comment:
As mentioned in the previous comment, we can change the return type to following

```suggestion
func NewOperatorConditionLister(indexer cache.Indexer) Lister[operatorv2.OperatorCondition] {
```